pkg/controller/v1alpha1/llmisvc/router.go

```diff
@@ -0,0 +1,214 @@
+	ctx = log.IntoContext(ctx, logger)
+
+	logger.Info("Reconciling Router")
+
+	if err := r.reconcileScheduler(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("SchedulerReconcileError", "Failed to reconcile scheduler: %v", err.Error())
+		return fmt.Errorf("failed to reconcile scheduler: %w", err)
+	}
+
+	// We do not support Gateway's spec, when creating HTTPRoutes either the default gateway or those provided
+	// as refs are attached to reconciled routes
+	if err := r.reconcileHTTPRoutes(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("HTTPRouteReconcileError", "Failed to reconcile HTTPRoute: %v", err.Error())
+		return fmt.Errorf("failed to reconcile HTTP routes: %w", err)
+	}
+
+	if err := r.reconcileIstioDestinationRules(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("IstioDestinationRuleReconcileError", "Failed to reconcile DestinationRule: %v", err.Error())
+		return fmt.Errorf("failed to reconcile istio destination rules: %w", err)
+	}
+
+	llmSvc.MarkRouterReady()
+
+	return nil
+}
+
+func (r *LLMISVCReconciler) reconcileHTTPRoutes(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling HTTPRoute")
+
+	expectedHTTPRoute := r.expectedHTTPRoute(ctx, llmSvc)
+
+	// TODO should we remove "llmSvc.Spec.Router.Route.HTTP == nil" from the condition below so that a non nil Route meeans "all type of routes are enabled"?
+	if llmSvc.Spec.Router == nil || llmSvc.Spec.Router.Route == nil || llmSvc.Spec.Router.Route.HTTP == nil {
+		return Delete(ctx, r, llmSvc, expectedHTTPRoute)
+	}
+
+	referencedRoutes, err := r.collectReferencedRoutes(ctx, llmSvc)
+	if err != nil {
+		return fmt.Errorf("failed to collect referenced routes: %w", err)
+	}
+
+	route := llmSvc.Spec.Router.Route
+
+	if route.HTTP.HasRefs() {
+		return Delete(ctx, r, llmSvc, expectedHTTPRoute)
+	}
+
+	// TODO(validation): referenced gateway exists
+	if route.HTTP.HasSpec() {
+		if err := Reconcile(ctx, r, llmSvc, &gwapiv1.HTTPRoute{}, expectedHTTPRoute, semanticHTTPRouteIsEqual); err != nil {
+			return fmt.Errorf("failed to reconcile HTTPRoute %s/%s: %w", expectedHTTPRoute.GetNamespace(), expectedHTTPRoute.GetName(), err)
+		}
+		referencedRoutes = append(referencedRoutes, expectedHTTPRoute)
+	}
+
+	return r.updateRoutingStatus(ctx, llmSvc, referencedRoutes...)
+}
+
+func (r *LLMISVCReconciler) collectReferencedRoutes(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) ([]*gwapiv1.HTTPRoute, error) {
+	if llmSvc.Spec.Router == nil || llmSvc.Spec.Router.Route == nil || !llmSvc.Spec.Router.Route.HTTP.HasRefs() {
+		return nil, nil
+	}
+
+	referencedRoutes := make([]*gwapiv1.HTTPRoute, 0, len(llmSvc.Spec.Router.Route.HTTP.Refs))
+	for _, routeRef := range llmSvc.Spec.Router.Route.HTTP.Refs {
+		route := &gwapiv1.HTTPRoute{}
+		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: routeRef.Name, Name: llmSvc.GetNamespace()}, route); err != nil {
```

Comment:
Incorrect assignment?

```suggestion
		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: llmSvc.GetNamespace(), Name: routeRef.Name}, route); err != nil {
```

Reply:
Done

Reply:
Done